File: src/main/java/tk/skulk/plugin/core/extensions/messageoverride/listeners/PlayerAdvancementDoneListener.java
package tk.skulk.plugin.core.extensions.messageoverride.listeners;

import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.player.PlayerAdvancementDoneEvent;
import org.jetbrains.annotations.NotNull;
import tk.skulk.plugin.core.extensions.messageoverride.MessageOverrideExtension;
import tk.skulk.plugin.wrapper.BaseListener;

import static tk.skulk.plugin.helpers.MiniMessageHelper.makeMessageWithComponent;


// FIXME
public final class PlayerAdvancementDoneListener extends BaseListener<MessageOverrideExtension> {
    public PlayerAdvancementDoneListener(final @NotNull MessageOverrideExtension extension) {
        super(extension);
    }

    @SuppressWarnings("ConstantConditions")
    @EventHandler(priority = EventPriority.LOWEST, ignoreCancelled = true)
    public void onPlayerAdvancementDone(final @NotNull PlayerAdvancementDoneEvent event) {
        event.message(makeMessageWithComponent("blue", '✓', "<0>.", event.message()));
    }
}

File: src/main/kotlin/tk/skulk/plugin/core/extensions/messageOverride/listeners/PlayerAdvancementDoneListener.kt
package tk.skulk.plugin.core.extensions.messageOverride.listeners

import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.player.PlayerAdvancementDoneEvent
import tk.skulk.plugin.core.SNPlugin
import tk.skulk.plugin.core.extensions.messageOverride.MessageOverrideExtension
import tk.skulk.plugin.framework.Listener
import tk.skulk.plugin.util.makeMessageWithComponents

// FIXME: Doesn't work for some reason.
class PlayerAdvancementDoneListener(extension: MessageOverrideExtension) : Listener<SNPlugin, MessageOverrideExtension>(
    extension
) {
    @EventHandler(priority = EventPriority.LOWEST, ignoreCancelled = true)
    fun onPlayerAdvancementDone(event: PlayerAdvancementDoneEvent) {
        event.message(makeMessageWithComponents("blue", '✓', "<0>.", event.message()!!))
    }
}
